cinder/backup/chunkeddriver.py

```diff
@@ -778,26 +726,51 @@ def delete_backup(self, backup):
 
 class BackupRestoreHandle(object):
     """Class used to reconstruct a backup from chunks."""
-    def __init__(self, chunked_driver):
+    def __init__(self, chunked_driver, volume_id, volume_file):
         self._driver = chunked_driver
+        self._volume_id = volume_id
+        self._volume_file = volume_file
         self._segments = []
         self._object_readers = {}
-
-    def __iter__(self):
         self._idx = -1
-        return self
 
-    # Python2 compatibility, __next__ is for Python3.
-    def next(self):
-        return self.__next__()
+    def add_backup(self, backup, metadata):
```

Comment:
D you know about the `abc` module or are you explicitly not enforcing the interface? If it's the second, then please document it in the function body, why this raises an exception by default.

Reply:
Good point. Made it abstract.